Source/Knowledge/CompKnowledge.cs

```diff
@@ -259,7 +259,7 @@ public void AcquireWeaponKnowledge(FactionDef faction)
                         }
                         if (Prefs.LogVerbose) stringBuilder.Append(pawn.gender.GetPossessive().ToLower() + " faction's tech level");
                     }
-                    if (isPlayer && ModBaseHumanResources.WeaponPoolIncludesScenario && !ModBaseHumanResources.FreeScenarioWeapons)
+                    if (isPlayer && ModBaseHumanResources.WeaponPoolIncludesScenario && !(ModBaseHumanResources.FreeScenarioWeapons || ModBaseHumanResources.unlocked.knowAllStartingWeapons))
```

Comment:
Yes, I know that the Code works as far as a Player is concerned, but the inside of that if Block has no use, which is why you already added the `!ModBaseHumanResources.FreeScenarioWeapons` in 5728c82b4d2f4e505f19dca16bfae1296fccbf8b . 
This change is only here to make all Effects the same, whether you use the setting or the Scenario Rule.

As for ranged vs. melee, I don't think is needed and could be even detrimental, as most/all of the Weapons in the patched scenarios are ranged Weapons.

But this got me an Idea for a possible new Option, but i don't know how easy it would be to implement:
If after all starting Pawns got their Weopans assigned, there still exists at least one Weapon without any possible user, add the knowledge to the Pawn with the highest Melee/Shooting Skill.

Reply:
We'd need to monitor the weapons added by the scenario before the pawns spawn, and then check them off one by one when pawns spawn. Also, it would be necessary to choose which pawns get which weapon ... not soo difficult but also not somethign trivial. I'm happy with the method of allowing them all weapons at once. :-)